torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -595,10 +650,48 @@ def _annotate_output_for_int8_in_int8_out_pattern(
                     input_act = maxpool_node.args[0]
                     assert isinstance(input_act, Node)
                     assert isinstance(maxpool_node, Node)
-                    edge_or_node: Tuple[Node, Node] = (input_act, maxpool_node)
+                    edge_or_node = (input_act, maxpool_node)
                     getitem_node.meta[
                         "quantization_annotation"
                     ].output_qspec = SharedQuantizationSpec(edge_or_node)
+            elif node.target is torch.ops.aten.cat.default:
+                cat_node = node
+                cat_quantization_annotation = (
+                    cat_node.meta["quantization_annotation"]
+                    if "quantization_annotation" in cat_node.meta
+                    else None
+                )
+                if (
+                    cat_quantization_annotation
+                    and cat_quantization_annotation._is_output_of_quantized_pattern
+                ):
+                    # Annotate the output_qspec of cat_node
+                    first_input_node = cat_node.all_input_nodes[0]
+                    assert isinstance(first_input_node, Node)
+                    assert isinstance(cat_node, Node)
+                    edge_or_node = (first_input_node, cat_node)
+                    cat_node.meta[
+                        "quantization_annotation"
+                    ].output_qspec = SharedQuantizationSpec(edge_or_node)
+            elif node.target is torch.ops.aten.avg_pool2d.default:
+                avg_pool2d_node = node
+                avg_pool2d_quantization_annotation = (
+                    avg_pool2d_node.meta["quantization_annotation"]
+                    if "quantization_annotation" in avg_pool2d_node.meta
+                    else None
+                )
+                if (
+                    avg_pool2d_quantization_annotation
+                    and avg_pool2d_quantization_annotation._is_output_of_quantized_pattern
+                ):
+                    # Annotate the output_qspec of cat_node
```

Comment:
This comment is not aligned with avg_pool2d you are handling. Also, is it possible to factor out the common code for annotating ops with `SharedQuantizationSpec`?

Reply:
Thanks for comment. Yeap, refactor the common code for `cat` and `avg_pool2d`. As for `max_pool2d_with_indices`, keep it as it since it's slightly different with 2 nodes in the pattern actually.